From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Mon, 18 Mar 2019 10:42:15 +0100
Subject: [PATCH] don't add rpaths for build directories when cross-compiling

The binaries cannot be executed anyways, so this just increases the binary
size unnecessarily.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 mesonbuild/linkers/linkers.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/mesonbuild/linkers/linkers.py b/mesonbuild/linkers/linkers.py
index c4df0fa1d893..50e11175e413 100644
--- a/mesonbuild/linkers/linkers.py
+++ b/mesonbuild/linkers/linkers.py
@@ -677,7 +677,10 @@ class GnuLikeDynamicLinkerMixin(DynamicLinkerBase):
         # Need to deduplicate rpaths, as macOS's install_name_tool
         # is *very* allergic to duplicate -delete_rpath arguments
         # when calling depfixer on installation.
-        all_paths = mesonlib.OrderedSet([os.path.join(origin_placeholder, p) for p in processed_rpaths])
+        if not env.is_cross_build():
+            all_paths = mesonlib.OrderedSet([os.path.join(origin_placeholder, p) for p in processed_rpaths])
+        else:
+            all_paths = mesonlib.OrderedSet()
         rpath_dirs_to_remove: T.Set[bytes] = set()
         for p in all_paths:
             rpath_dirs_to_remove.add(p.encode('utf8'))
@@ -705,7 +708,8 @@ class GnuLikeDynamicLinkerMixin(DynamicLinkerBase):
                 paths = padding
             else:
                 paths = paths + ':' + padding
-        args.extend(self._apply_prefix('-rpath,' + paths))
+        if paths:
+            args.extend(self._apply_prefix('-rpath,' + paths))
 
         # TODO: should this actually be "for solaris/sunos"?
         # NOTE: Remove the zigcc check once zig support "-rpath-link"
